### ISSUE TITLE
Missing documentation?

### ISSUE THREAD

**Issue by larsbrinkhoff** [pipcet/lbForth#2]
Now that you have poked a bit at lbForth, which dark corners do you think would benefit most from better documentation?

**Comment by pipcet** [comment]
One area that still confuses me is the use of vocabularies in meta.fth. We sometimes seem to expect `include`d files to change the current vocabulary, and I can't, or couldn't, quite figure out what those expectations where when I was working on the wasm backend.

(I'd generally err on the side of commenting the metacompiler more liberally, since comments there don't really hurt anything.)

**Comment by larsbrinkhoff** [comment]
> One area that still confuses me is the use of vocabularies in meta.fth.

Right, it took me three attempts to figure it out myself until I finally had a working metacompiler.

> We sometimes seem to expect included files to change the current vocabulary

I think the included files are supposed not to change the state of the search order.

Do you mean how the assembler juggles the vocabulary?
